libr/core/cmd_search.c

```diff
@@ -230,6 +231,8 @@ R_API int r_core_search_prelude(RCore *core, ut64 from, ut64 to, const ut8 *buf,
 	int ret;
 	ut64 at;
 	ut8 *b = (ut8 *)malloc (core->blocksize);
+	int timeout = r_config_get_i (core->config, "anal.aaptimeout");
+	time_t stop_time = time(NULL) + timeout;
```

Comment:
in addition , having an api for this can allow us to globally enable/disable the timeout functionality, aoviding the need to call time() all the time in a loop that will be a heavy performance penatly in many situations that is not needed